File: 코테 실습/파이썬/A로 B 만들기.py
def solution(before, after):
    # 두 문자열을 각각 정렬해서 비교.
    # 애너그램이라면 정렬했을 때 완전히 같은 문자열이 됨.
    if sorted(before) == sorted(after):
        return 1
    else:
        return 0
    
    # 한 줄로도 가능: return 1 if sorted(before) == sorted(after) else 0

'''
아래 내용도 괜찮음 참고해보기
'''

def solution(before, after):
    
    after_list = list(after) 
    
    for char_b in before:
        if char_b in after_list:
            after_list.remove(char_b)
        else:
            return 0
            
    # 아무것도 없을 때 false, 있다면 true
    if not after_list: # after_list가 비어있으면 True
        return 1
    else: # after_list가 비어있지 않으면 False
        return 0

File: 코테 실습/JS/A로 B 만들기.js
/*
splice() 메소드의 기본적인 형태는 배열.splice(startIndex, deleteCount, item1, item2, ...) 이렇게 돼.
startIndex: 작업을 시작할 배열의 인덱스.
deleteCount: startIndex부터 몇 개의 요소를 제거할지 지정. (0이면 제거하지 않음)
item1, item2, ...: (선택 사항) 제거된 요소 자리에 추가할 요소들.
*/

function solution(before, after) {
    answer = 0;

    let afterArray = after.split(''); 
    for (let i = 0; i < before.length; i++) {
        const char_b = before[i];
        const indexInAfter = afterArray.indexOf(char_b);

        if (indexInAfter !== -1) {
            afterArray.splice(indexInAfter, 1);
        } else {
            return 0; 
        }
    }

    if (afterArray.length === 0) {
        return 1;
    } else {
        return 0;
    }
}